src/system_contract.rs

```diff
@@ -25,10 +28,38 @@ pub enum SystemContract {
     /// * Transaction::keys[0] - source
     /// * Transaction::keys[1] - destination
     Move { tokens: i64 },
+    /// Load a contract
+    /// contract_id - id to associate this contract
+    /// nanme - file path of the contract to load
+    Load { contract_id: Pubkey, name: String },
+    /// Call a contract
+    /// contract_id - contract to call
+    /// data - contract specific data
+    Call { contract_id: Pubkey, data: Vec<u8> },
 }
 
 pub const SYSTEM_CONTRACT_ID: [u8; 32] = [0u8; 32];
 
+#[derive(Debug)]
+pub struct AccountInfo<'a> {
+    pub key: &'a Pubkey,
+    pub account: &'a mut Account,
+}
+
+// TODO where should this live?
+struct Temporary {
+    pub loaded_contracts: RwLock<HashMap<Pubkey, DynamicContract>>,
+}
+
+impl Temporary {
+    pub fn new() -> Self {
+        Temporary {
+            loaded_contracts: RwLock::new(HashMap::new()),
+        }
+    }
+}
+static mut TEMPORARY: Option<Temporary> = None;
```

Comment:
Can we drag this data structure around as a function parameter or struct field instead?

Reply:
Planning to move it to bank where it becomes since per @mvines comment we will be calling the entrypoint from the bank and not via the system_contract